generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs

```diff
@@ -42,6 +42,21 @@ if (reactive) {
     callBlock = ".block()";
     callListBlock = ".collectList().block()";
 }
+let saveMethod = 'save';
+if (!reactive && databaseType === 'sql') {
+    saveMethod = 'saveAndFlush';
+}
+let idValue = `${asEntity(entityInstance)}.getId()`;
+if (databaseType === 'sql' && hasOauthUser !== true) {
+    idValue = idValue + '.intValue()';
+} else if (databaseType === 'cassandra') {
+    idValue = idValue + '.toString()';
+}
```

Comment:
Could you replace these conditions by the following ones, that are more logical and direct (I know this was like that in previous code, but it's a good opportunity to improve it)
```suggestion
if (primaryKeyType === 'Long') {
    idValue = idValue + '.intValue()';
} else if (primaryKeyType === 'UUID') {
    idValue = idValue + '.toString()';
}
```

Reply:
Good idea!